content/news/XXXX-XX-XX-dvs-internals-pt2.md

```diff
@@ -0,0 +1,301 @@
+
+![Serato Timecode Signal]({static}/images/news/timecode-signal.png)
+
+If we look closely, we notice that the peaks occur every time the right channel crosses zero and the signal value goes from the positive part of the signal to negative part.
+
+This is very convenient, because that way we don't have to lose any sleep over how to detect the peaks in the left channel, we just detect if there was a positive-to-negative zero crossing in the *right* channel, then take the current value of the left one and check if it's a `0` or `1` by comparing it with a threshold.
+
+Now it's clear how the analog timecode signal can be converted to a stream of bits.
+But how can these bits be interpreted as positions?
+Or rather: How did the engineers that created the timecode format encode the positions as bits?
+
+## Encoding strategies
+### Simple Base-2 Encoding
+
+The intuitive approach is to encode positions as a sequence of ascending numbers, e.g. the first position is 0, the second position is 1, the third position is 2, etc.
+Since we're working with bits, we could simply express these decimal numbers in the base-2 (binary) numeral system, like this:
+
+| Position | Bit Sequence |
+| -------- | ------------ |
+| 0        | `000`        |
+| 1        | `001`        |
+| 2        | `010`        |
+| 3        | `011`        |
+| 4        | `100`        |
+| 5        | `101`        |
+| 6        | `110`        |
+| 7        | `111`        |
+
+Of course, that is just a toy example, but even if we only had 8 different positions, we need use 3 bits (= 3 cycles) to represent each position.
+In reality, we want to encode *a lot* more different positions, which means the resulting binary number will be longer, meaning we need more bits and therefore cycles per position we want to encode.
+
+The corresponding signal would look like this:
+
+![Simple bitstream with 3 bits per position]({static}/images/news/timecode-3-bit-bitstream.svg)
+
+The first 3 cycles encode position 0, the next 3 cycles encode position 1, and so on.
+Simple, right?
+Unfortunately, this approach has a major problem.
+
+Let's assume you seek to a random position in the track (by picking up the needle and dropping it somewhere) and the
+next 3 bits are `001`.
+Do you know what position or cycle number we are at?
+
+We can look up `001` in the table above and see that these bits encode position 1 (cycles 3-6).
+So after reading these bits we should be at cycle 6.
+And that is *one* possibility, but not the only one.
+
+![Simple bitstream with 3 bits per position and all occurrences of 001 highlighted]({static}/images/news/timecode-3-bit-bitstream-highlight.svg)
+
+So there are multiple possible possible positions and we don't know which one is the correct one.
+We can mitigate this by reading the next 3 cycles, too, but this doubles the amount of cycles we have to read (6 instead of 3 cycles) until we know the position for sure.
```

Comment:
```suggestion
We can mitigate this by reading the next 3 cycles too, but this doubles the amount of cycles we have to read (6 instead of 3 cycles) until we know the position for sure.
```